src/components/sidebar/right-sidebar/datepicker/DateSlider.tsx

```diff
@@ -0,0 +1,97 @@
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import './DateSlider.css';
+
+interface DSprops {
+    min: number;
+    max: number;
+    values: Date[];
+    minDate: Date;
+    onChange: Function;
+}
+
+export default function DateSlider(props: DSprops){
+
+    const {min, max, minDate} = props;
+    const [values, setValues] = useState<number[]>([(props.values[0].getTime() - minDate.getTime())/86400000,
+        (props.values[1].getTime() - minDate.getTime())/86400000])
+    const [minVal, setMinVal] = useState(values[0]);
+    const [maxVal, setMaxVal] = useState(values[1]);
+    const minValRef = useRef(min);
+    const maxValRef = useRef(max);
+    const range = useRef<HTMLDivElement>(null);
+
+    const getPercent = useCallback(
```

Comment:
just curious & trying to learn more, why are we using useCallback here???

Reply:
useCallBack makes sure that the function is only redefined when one of the deps changes, instead of redefining it every rerender. So getPercent would be redefined if min or max changed (which are passed in as props). which would then trigger the useEffect causing the min and max percent to change moving the rail bar to the right position. by using useCallBack we cause a rerender or recalculation if and only if the min and max values of the whole bar change which is highly unlikely (possible but rare). and so this improves performance by reducing rerendering. 

That being said, I only really leart what useCallBack does while going through the tutorial for the slider and then looked it up a little more to understand, and deduced the above reasoning which makes sense to me. SO if you feel like I might be misunderstanding its use or that the performance gain really is not that great, I am open to criticism.

Reply:
Also all that being said, the more I am a little more confused as to whether it is needed here or not. @brett-stephen @annoyingpr if you guys have any input on this, it is more than welcome as well.